### DISCUSSION TITLE
Definition of NER model metrics

### DISCUSSION THREAD

**Discussion by joaoaleite** [explosion/spaCy#8111] [Help: Coding & Implementations] [answered]
Hi,

I was training ner models with spacy 3.0 and I was wondering what are the precision, recall and f-score metrics exacly. [This blog](http://www.davidsbatista.net/blog/2018/05/09/Named_Entity_Evaluation/) mentions two ways of computing those metrics. The first one follows the CoNLL-2003 definition, which only cares about exact matching of the entities.

“precision is the percentage of named entities found by the learning system that are correct. Recall is the percentage of named entities present in the corpus that are found by the system. A named entity is correct only if it is an exact match of the corresponding entity in the data file.”

The second definition is from SemEval’13, which takes into consideration partial matches.

Which method is being used by spaCys 3.0 train interface? And is the other method implemented somewhere that I can use?

Thanks!

**Comment by davesuketu215** [comment]
Hi,

I feel you should have a look at this API - language.py --> evaluate( ). This invokes the spaCy scorer APIs. The scorer APIs are present in a file scorer.py
And more specifically, for NER, there is a function inside scorer.py, by the name --> get_ner_prf( )
Hope that helps.

**Comment by adrianeboyd** [comment, marked as answer]
The evaluation only counts entities that match exactly as correct. We don't have any other scoring methods for entities implemented.
